manifests/init.pp

```diff
@@ -73,28 +67,27 @@
     validate_absolute_path($install_dir)
   }
 
-  if $package_version == undef and versioncmp("${::clientversion}", '4.0.0') >= 0 {
-    info('puppet_agent performs no actions if a package_version is not specified on Puppet 4')
-  } elsif $package_version == undef and $is_pe {
-    info("puppet_agent performs no actions if the master's agent version cannot be determed on PE 3.x")
+  if $package_version == undef {
+    info('puppet_agent performs no actions if a package_version is not specified')
   } elsif defined('$::pe_server_version') {
     info('puppet_agent performs no actions on PE infrastructure nodes to prevent a mismatch between agent and PE components')
   } else {
-    if $package_version != undef and $package_version !~ /^\d+\.\d+\.\d+([.-]?\d*|\.\d+\.g[0-9a-f]+)$/ {
+    # In this code-path, $package_version != undef AND we are not on a PE infrastructure
+    # node since $::pe_server_version is not defined
+
+    if $package_version !~ /^\d+\.\d+\.\d+([.-]?\d*|\.\d+\.g[0-9a-f]+)$/ {
       fail("invalid version ${package_version} requested")
     }
 
     # Strip git sha from dev builds
-    if ($package_version != undef and $package_version =~ /g/){
+    if $package_version =~ /g/ {
       $_expected_package_version = split($package_version, /[.-]g.*/)[0]
     } else {
       $_expected_package_version = $package_version
     }
 
     $aio_upgrade_required = (getvar('::aio_agent_version') == undef) or
```

Comment:
I wonder if I can remove the `(getvar('::aio_agent_version') == undef)` check? That fact should be defined on Puppet 4 agents, shouldn't it?

Reply:
It should blow up a few lines further down if it's missing.

Reply:
Turns out I can remove it -- checking if it is undef is a quick way to see if you're running on a pre-Puppet 4 node.